DllImportGenerator/DllImportGenerator/Analyzers/ManualTypeMarshallingAnalyzer.cs

```diff
@@ -419,26 +463,17 @@ private void AnalyzeNativeMarshalerType(SymbolAnalysisContext context, ITypeSymb
 
                         context.ReportDiagnostic(Diagnostic.Create(NativeTypeMustBePointerSizedRule,
                             valueProperty is not null
-                                ? GetSyntaxReferenceForDiagnostic(valueProperty).GetSyntax().GetLocation()
-                                : GetSyntaxReferenceForDiagnostic(nativeType).GetSyntax().GetLocation(),
+                                ? valueProperty.Locations[0]
```

Comment:
These checks are under a branch with several previously performed checks. Is there a better way to express this algorithm - I see a lot of complicated rebranching logic that I hope can be reevaluated.

Reply:
I've tried to simplify these checks a bit. Let me know what you think.